projects/admin/src/app/circulation/patron-transaction.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* tslint:disable */
+// required as json properties is not lowerCamelCase
+
+import * as moment from 'moment';
+import { marker } from "@biesbjerg/ngx-translate-extract-marker";
+
+export function _(str) {
+  return marker(str);
+}
+
+export enum PatronTransactionStatus {
+  OPEN = _('open'),
+  CLOSED = _('closed')
+}
+export enum PatronTransactionEventType {
+  FEE = _('fee'),
+  PAYMENT = _('payment'),
+  DISPUTE = _('dispute'),
+  RESOLVED = _('resolved')
+}
+
+export class PatronTransaction {
+
+  pid: string;
+  creation_date: string;
+  type: string;
+  status: PatronTransactionStatus = PatronTransactionStatus.OPEN;
+  total_amount: number = 0;
+  events: Array<PatronTransactionEvent> = [];
+  note?: string = undefined;
+  document?: any = undefined;
+  loan?: any = undefined;
```

Comment:
maybe some comments

Reply:
As this is just a wrapping class corresponding to RERO_ILS schema and according to other 'class' classes already present into the project, I just create comments for method classes.